### PR TITLE
BUG: Filter/transform fail in some cases when multi-grouping with a datetime-like key

### PR DESCRIPTION
**Pull request by evanpw**
Fixes GH #10114, and a related problem in transform which was already present in 0.16.0

### CODE REVIEW

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -598,7 +610,7 @@ def get_group(self, name, obj=None):
         if obj is None:
             obj = self._selected_obj
 
-        inds = self._get_index(name)
+        inds = self._get_index(name, raise_on_missing=True)
```
Comment (jreback) [inline pandas/core/groupby.py]: I think I would drop the ``raise_on_missing`` argument and return the default of ``[self.indices.get(name, []) for name in names]`` in ``_get_index``, then for here:

```
if not len(inds):
    raise KeyError(name)
```

Comment (evanpw) [inline pandas/core/groupby.py]: I was a little worried that there could be a key in self.indices which has an empty array as its value (like a categorial grouper with not all categories represented), but it looks like that's impossible.

**Comment by jreback** [comment]
can you rebase

File: doc/source/whatsnew/v0.16.2.txt [hunk 2]
```diff
@@ -65,6 +65,8 @@ Bug Fixes
 - Bug in ``Timestamp``'s' ``microsecond``, ``quarter``, ``dayofyear``, ``week`` and ``daysinmonth`` properties return ``np.int`` type, not built-in ``int``. (:issue:`10050`)
 - Bug in ``NaT`` raises ``AttributeError`` when accessing to ``daysinmonth``, ``dayofweek`` properties. (:issue:`10096`)
 
+- Bug in ``filter`` (regression from 0.16.0) and ``transform`` when grouping on multiple keys, one of which is datetime-like (:issue:`10114`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.2.txt]: pls move to 0.17.0

**Comment by jreback** [comment]
can you rebase and ping, will have a look

**Comment by evanpw** [comment]
Rebased

**Comment by jreback** [comment]
thanks! keep-em coming!

**Comment by evanpw** [comment]
Thanks!

**Pull request merged**